src/IRremoteESP8266.h

```diff
@@ -645,7 +652,8 @@
      DECODE_DAIKIN216 || DECODE_SHARP_AC || DECODE_DAIKIN160 || \
      DECODE_NEOCLIMA || DECODE_DAIKIN176 || DECODE_DAIKIN128 || \
      DECODE_AMCOR || DECODE_DAIKIN152 || DECODE_MITSUBISHI136 || \
-     DECODE_MITSUBISHI112 || DECODE_HITACHI_AC424 || DECODE_HITACHI_AC3)
+     DECODE_MITSUBISHI112 || DECODE_HITACHI_AC424 || DECODE_HITACHI_AC3 || \
```

Comment:
Not releated to this commit, but is the recently added Carrier AC protocol missing here?
Can we automate this in some way as to not miss that, or maybe just have a checker for it? (or is it correct to not add them here?, could it be documented?)

Reply:
Protocols only need to be added here **if** they use the `state` part of the `decode_results` class/structure.
e.g.
https://github.com/crankyoldgit/IRremoteESP8266/blob/494c5957626d3c92bbfd87bb73d3f04569cbd023/src/IRrecv.h#L103
https://github.com/crankyoldgit/IRremoteESP8266/blob/494c5957626d3c92bbfd87bb73d3f04569cbd023/src/IRrecv.h#L58-L64

Basically, it's just a compile time memory optimisation for when someone compiles the library without any protocols that use `state`. ie. Saves about 50 bytes! (I know, YUGE!)

p.s. The answer to "Should this be documented (better)?" Is always, "Yes!. PR's welcome ;-)

Reply:
> Can we automate this in some way as to not miss that, or maybe just have a checker for it? (or is it correct to not add them here?, could it be documented?)

We could probably write an external tests for it (python/shell etc) and add it to Travis etc.
Or we could waste the 50+ bytes per copy of the class, and just make it always on and simplify things.

Reply:
50 bytes are huge at times!
Then I know and understand it better, I will read a bit more and maybe come back with suggestions